data-raw/temperature-data/temp_regression_vignette.Rmd

```diff
@@ -0,0 +1,797 @@
+summary(feather_lfc_mod_mean)
+test_predict <- predict(feather_lfc_mod_mean, test)
+test_predict_df <- test |>
+  mutate(predicted = test_predict)
+# evaluate predictions - MAPE of 10% is not bad
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Building Min Regression
+
+  * Splitting data
+  * Building the Model
+  * Model Summary
+  * Making predictions
+  * Evaluating predictions
+  
+```{r}
+# MIN Regression
+split <-rsample::initial_split(feather_lfc_regression_data_min, prop = 0.8)
+train <- rsample::training(split)
+test <- rsample::testing(split)
+feather_lfc_mod_min <- lm(temp ~ date + butte_temp, data = train)
+summary(feather_lfc_mod_min)
+test_predict <- predict(feather_lfc_mod_min, test)
+test_predict_df <- test |>
+  mutate(predicted = test_predict)
+# evaluate predictions - MAPE of 10% is not bad
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Building Max Regression
+
+  * Splitting data
+  * Building the Model
+  * Model Summary
+  * Making predictions
+  * Evaluating predictions
+  
+```{r}
+# MAX Regression
+split <-rsample::initial_split(feather_lfc_regression_data_max, prop = 0.8)
+train <- rsample::training(split)
+test <- rsample::testing(split)
+feather_lfc_mod_max <- lm(temp ~ date + butte_temp, data = train)
+summary(feather_lfc_mod_max)
+test_predict <- predict(feather_lfc_mod_max, test)
+test_predict_df <- test |>
+  mutate(predicted = test_predict)
+# evaluate predictions - MAPE of 10% is not bad
+mean(abs((
+  test_predict_df$predicted - test_predict_df$temp
+)) / test_predict_df$temp)
+```
+
+#### Predictions
+
+  * Combine Predictions (mean, min, max dataframes into one)
+  * Reshape data
+
+```{r}
+# Predictions
+feather_gap_predicted_mean_lfc <- predict(feather_lfc_mod_mean, feather_gap_mean)
+feather_lfc_mean_predicted <- feather_gap_mean |> 
+  mutate(value = feather_gap_predicted_mean_lfc,
+         statistic = "mean") |> 
+  select(date, value, statistic)
+ggplot(feather_lfc_mean_predicted, aes(x = date, y = value)) +
+  geom_line()
+feather_gap_predicted_min_lfc <- predict(feather_lfc_mod_min, feather_gap_min)
+feather_lfc_min_predicted <- feather_gap_min |> 
+  mutate(value = feather_gap_predicted_min_lfc,
+         statistic = "min") |> 
+  select(date, value, statistic)
+ggplot(feather_lfc_min_predicted, aes(x = date, y = value)) +
+  geom_line()
+feather_gap_predicted_max_lfc <- predict(feather_lfc_mod_max, feather_gap_max)
+feather_lfc_max_predicted <- feather_gap_max |> 
+  mutate(value = feather_gap_predicted_max_lfc,
+         statistic = "max") |> 
+  select(date, value, statistic)
+ggplot(feather_lfc_max_predicted, aes(x = date, y = value)) +
+  geom_line()
+```
+
+#### Full dataset
+
+  * Join with original data (merge combined predictions with the original dataset that includes gage agency and gage number)
+  * Visualize combined data
+
```

Comment:
I don't think we need to show all this code.

Also should include some interpretation of the plots